### PR TITLE
Doughnut Chart Component

### PR DESCRIPTION
**Pull request by khamis-hesham**
Relates #65 
- [x] installed react-chartjs-2 and chart.js. [from here](https://www.npmjs.com/package/react-chartjs-2)
- [x]  create doughnut chart component . [docs](https://www.chartjs.org/docs/latest/charts/doughnut.html)
- [x] convert react class component to functional component with react Hooks.

### CODE REVIEW

File: client/src/components/Chart.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+              position:this.props.legendPosition
+            }
+          }}
+        />
+      </div>
+    );
+  }
+}
+
+export default Chart;
```
Comment (lina-jamal) [inline client/src/components/Chart.js:38]: why you push the class component?
you do not need it after you converted the class component to a function component

File: client/src/App.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
 import "./App.css";
+import Chart from "./components/Chart"
 
 function App() {
   return (
     <div className="App">
       <h1> Hi from React App</h1>
+      <Chart />
```
Comment (lina-jamal) [inline client/src/App.js:8]: after you testing the component you must remove it from App.js

**Review by lina-jamal — COMMENTED** [review]
resolve the conflicts

File: client/.eslintcache [hunk 3]
```diff
@@ -1 +1 @@
...
...:"5"},{"filePath":"8","messages":"9","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"10"},"serryl",{"filePath":"11","messages":"12","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"10"},{"filePath":"13","messages":"14","errorCount":1,"warningCount":1,"fixableErrorCount":0,"fixableWarningCount":0,"source":null},"/home/khamis/Desktop/geeks-G9/calorie-tracker/client/src/index.js",[],["15","16"],"/home/khamis/Desktop/geeks-G9/calorie-tracker/client/src/App.js",[],"/home/khamis/Desktop/geeks-G9/calorie-tracker/client/src/components/Chart.js",["17","18"],{"ruleId":"19","replacedBy":"20"},{"ruleId":"21","replacedBy":"22"},{"ruleId":"23","severity":1,"message":"24","line":2,"column":10,"nodeType":"25","messageId":"26","endLine":2,"endColumn":18},{"ruleId":"27","severity":2,"message":"28","line":7,"column":15,"nodeType":"25","messageId":"29","endLine":7,"endColumn":20},"no-native-reassign",["30"],"no-negated-in-lhs",["31"],"no-unused-vars","'Doughnut' is defined but never used.","Identifier","unusedVar","no-undef","'props' is not defined.","undef","no-global-assign","no-unsafe-negation"]
```
Comment (hassanelnajjar) [inline client/.eslintcache:1]: please remove the eslintcache using git rm client/.eslintcache --cached and then add it to .gitignore

File: client/src/components/DoughnutChart.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  );
+};
+
+DoughnutChart.propTypes = {
+  legend: propTypes.arrayOf(propTypes.string).isRequired,
+  section: propTypes.arrayOf(propTypes.number).isRequired,
+  sectionBackground: propTypes.arrayOf(propTypes.string).isRequired,
+  width: propTypes.arrayOf(propTypes.number).isRequired,
+  height: propTypes.arrayOf(propTypes.number).isRequired,
+};
```
Comment (imansedky5) [inline client/src/components/DoughnutChart.js:42]: use destructuring

File: client/package.json [hunk 5]
```diff
@@ -6,7 +6,9 @@
     "@testing-library/jest-dom": "^5.11.9",
     "@testing-library/react": "^11.2.3",
     "@testing-library/user-event": "^12.6.2",
+    "chart.js": "^2.9.4",
```
Comment (hassanelnajjar) [inline client/package.json:9]: do we need to chart.js module, you install the react-chartjs-2 and use it at the Chart.js?

File: client/src/components/DoughnutChart.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import { Doughnut } from "react-chartjs-2";
+import propTypes from "prop-types";
```
Comment (hassanelnajjar) [inline client/src/components/DoughnutChart.js:2]: as a convention, the propTypes that import from the "prop-types" should be PropTypes
(The first Letter is Capital )

please see the [link](https://reactjs.org/docs/typechecking-with-proptypes.html)

File: client/src/components/DoughnutChart.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+          responsive: true,
+          maintainAspectRatio: true,
+        }}
+      />
+    </div>
+  );
+};
+
+DoughnutChart.propTypes = {
+  legend: propTypes.arrayOf(propTypes.string).isRequired,
```
Comment (hassanelnajjar) [inline client/src/components/DoughnutChart.js:37]: Please use destructuring of propTypes !

File: client/src/components/Chart.js (hunk 1 above)
Comment (khamis-hesham) [reply client/src/components/Chart.js:38]: Ok :+1:

**Review by zeinjendeya — CHANGES_REQUESTED** [review]
this branch still has conflicts, please make sure to resolve them.

File: client/.gitignore [hunk 8]
```diff
@@ -29,7 +29,11 @@ yarn-error.log*
 /node_modules
 /.pnp
 .pnp.js
+<<<<<<< HEAD
```
Comment (hassanelnajjar) [inline client/.gitignore:32]: Please edit the gitignore file

**Review by hassanelnajjar — DISMISSED** [review]

**Review by lina-jamal — APPROVED** [review]

**Review by imansedky5 — APPROVED** [review]

**Review by zeinjendeya — DISMISSED** [review]

**Review by hassanelnajjar — APPROVED** [review]

**Review by zeinjendeya — APPROVED** [review]

**Review by lina-jamal — APPROVED** [review]

**Review by imansedky5 — APPROVED** [review]

**Pull request merged**